gateway/gateway.go

```diff
@@ -470,7 +479,12 @@ func (gateway *HandleT) userWebRequestWorkerProcess(userWebRequestWorker *userWe
 			} else {
 				sourceStats[sourceTag].RequestEventsSucceeded(job.EventCount)
 			}
-			jobIDReqMap[job.UUID].done <- err
+			if reqOKMap[jobIDReqMap[job.UUID]] == "" {
+				reqOKMap[jobIDReqMap[job.UUID]] = err
+			}
+		}
+		for req, err := range reqOKMap {
+			req.done <- err
```

Comment:
can you explain the logic/motivation behind this?

Reply:
Each request can have multiple events.
We only want one message to be posted on the request's done channel, it being `err` if at least one event(now a `*jobsb.JobT`) fails storing. or an `OK` in case all events(`[]*jobsdb.JobT`) have stored successfully.

Reply:
Either all or no jobs from the batch should be stored in case of an error. We might require adaptations in `internalStoreWithRetryEachInTx` for this